SConstruct

```diff
@@ -1178,257 +1156,136 @@ if env['python_package'] == 'new':
-                    print("WARNING: The value for {0} has been set and will be used instead "
-                          "of {1}".format(key.format(major), key.format('')))
-                else:
-                    env[key.format(major)] = env[key.format('')]
-
-        del env['python_version']
-
-# Make sure everything gets converted to properly versioned variables by deleting
-# these so they don't get used accidentally
-del env['python_package']
-del env['python_array_home']
-del env['python_cmd']
-del env['python_prefix']
-
-require_python = any([env['python{}_package'.format(p)] == 'full' for p in ['2', '3']])
-want_python = any([env['python{}_package'.format(p)] == 'default' for p in ['2', '3']])
-
-if require_python or want_python:
-    # Check for Cython:
-    try:
-        import Cython
-        cython_version = LooseVersion(Cython.__version__)
-        assert cython_version >= cython_min_version
-    except ImportError:
-        message = "Cython could not be imported by the Python interpreter running SCons."
-        have_cython = False
-    except AssertionError:
-        message = ("Cython is an incompatible version: "
-                   "Found {0} but {1} or newer is required.".format(cython_version,
-                                                                    cython_min_version))
-        have_cython = False
-    else:
-        have_cython = True
-    finally:
-        if not have_cython:
-            if require_python:
-                print('ERROR: ' + message)
-                sys.exit(1)
-            elif want_python:
-                print('WARNING: ' + message)
-                for py_pkg in ['python{}_package'.format(p) for p in ['2', '3']]:
-                    if env[py_pkg] == 'default':
-                        env[py_pkg] = 'minimal-default'
-        else:
-            print('INFO: Using Cython version {0}.'.format(cython_version))
+env['install_python_action'] = ''
+env['python_module_loc'] = ''
 
-def configure_minimal_python(py_ver):
-    # Test to see if we can run Python if the minimal interface is to be built
+if env['python_package'] in ('full', 'default'):
+    # Test to see if we can import numpy and Cython
     warn_no_python = False
     script = textwrap.dedent("""\
-        from __future__ import print_function
         import sys
         print('{v.major}.{v.minor}'.format(v=sys.version_info))
+        err = ''
+        try:
+            import numpy
+            print(numpy.__version__)
+        except ImportError as np_err:
+            print('0.0.0')
+            err += str(np_err) + '\\n'
+        try:
+            import Cython
+            print(Cython.__version__)
+        except ImportError as cython_err:
+            print('0.0.0')
+            err += str(cython_err) + '\\n'
+        if err:
+            print(err)
     """)
 
+    if env['python_array_home']:
+        script = "sys.path.append({})\n".format(env['python_array_home']) + script
```

Comment:
I think if `python_array_home` this will result in a `NameError` because `sys` won't be imported when the `sys.path.append` is called. I think this is probably a mistake I made when I redid this a few months ago.

Reply:
Digging into it, it's from an earlier change: https://github.com/Cantera/cantera/commit/7d3ebdd3581c06f6294e4fb64e355007c2ea766e, which happens to be before 2.3.0 was released. And it's broken not only because `sys` hasn't been imported, but also because the path isn't quoted.

Which actually provides us with some useful information: nobody uses this option! I was even thinking about whether or not we could drop this, and this is definitive evidence that we can.

Reply:
Great! I was wondering if we could drop it too. I suppose its from a time when NumPy wasn't so easy to install, but that time is long gone.